examples/customresources/demos/complexmodule/moduletest/module_test.go

```diff
@@ -37,31 +38,45 @@ import (
 // needs to be added to the web service before it normally would be avalilable after completing
 // a config cycle.
 func TestComplexModule(t *testing.T) {
-	logger := logging.NewTestLogger(t)
-
-	// Modify the example config to run directly, without compiling the module first.
-	cfgFilename, port, err := modifyCfg(t, utils.ResolveFile("examples/customresources/demos/complexmodule/module.json"), logger)
-	test.That(t, err, test.ShouldBeNil)
-
-	serverPath, err := testutils.BuildTempModule(t, "web/cmd/server/")
-	test.That(t, err, test.ShouldBeNil)
-
-	// start the viam server with a temporary home directory so that it doesn't collide with
-	// the user's real viam home directory
-	testTempHome := t.TempDir()
-	server := pexec.NewManagedProcess(pexec.ProcessConfig{
-		Name:        serverPath,
-		Args:        []string{"-config", cfgFilename},
-		CWD:         utils.ResolveFile("./"),
-		Environment: map[string]string{"HOME": testTempHome},
-		Log:         true,
-	}, logger.AsZap())
-
-	err = server.Start(context.Background())
-	test.That(t, err, test.ShouldBeNil)
```

Comment:
Strictly speaking, this part of this test wouldn't cause an issue like we see with the others. If the port was taken, `server.Start` would return the failure to bind error.

But I figured if I was adding loops around the other tests to be robust, might as well get this one was well.